tools/heir-opt.cpp

```diff
@@ -119,7 +120,17 @@
 using namespace mlir;
 using namespace tosa;
 using namespace heir;
-using mlir::func::FuncOp;
+
+// hack here: another template specialization for FuncOp
+// expect linker to pick this one
```

Comment:
I'm really torn on this one - I love the IR it produces, but this *is* pretty hacky. At the very least, I'd say lets expand the comment here, maybe link it back to this PR or a new issue that explains why the other alternatives don't work.

Reply:
I'm OK with it, but I would be quick to remove it if it causes problems.

As long as no tests depend on the naming convention, it should be purely cosmetic.